core/src/main/java/inetsoft/util/cachefs/BinaryTransfer.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of StyleBI.
+ * Copyright (C) 2025  InetSoft Technology
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package inetsoft.util.cachefs;
+
+import inetsoft.util.FileSystemService;
+import inetsoft.util.Tool;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.DeferredFileOutputStream;
+import org.apache.commons.text.RandomStringGenerator;
+
+import java.io.*;
+import java.nio.file.*;
+
+public class BinaryTransfer implements Serializable {
```

Comment:
I'm not a big fan of having logic in a data structure class. I think it would be better to have `BinaryTransfer` be a pure data structure and have something like BinaryTransferService contain the logic.